x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -1566,6 +1568,18 @@ static void writeDateTrunc(PlanStreamOutput out, DateTrunc dateTrunc) throws IOE
         out.writeExpression(fields.get(1));
     }
 
+    static IpPrefix readIpPrefix(PlanStreamInput in) throws IOException {
+        return new IpPrefix(in.readSource(), in.readExpression(), in.readExpression());
+    }
+
+    static void writeIpPrefix(PlanStreamOutput out, IpPrefix ipPrefix) throws IOException {
+        out.writeSource(ipPrefix.source());
+        List<Expression> fields = ipPrefix.children();
+        assert fields.size() == 2;
+        out.writeExpression(fields.get(0));
+        out.writeExpression(fields.get(1));
```

Comment:
Could you put the body of this method on `IpPrefix` itself? That's how the rest of Elasticsearch works. ESQL works this way for a reason that we're in the process of making not true any more. The general pattern is
```
public IpPrefix(Source, Field, Field) { <-- the "normal" ctro
}

public IpPrefix(PlanStreamInput in) throws IOException { <-- the "streaming" ctor
}

public void writeTo(PlanStreamOutput out) throws IOException {
}
```

The important bit is that the streaming ctor and `writeTo` fit on the same screen. So we stick them next to each other. That way you can read them and use you eyes to check if they are a mirror of one another quickly.

Reply:
Moved them. However, I "couldn't" do the read constructor, because it created a conflict in the EsqlFunctionRegistry, where it couldn't find the correct constructor. I think I could have fixed it with explicit generics or something like that, but it would be extra things that I'm not sure would be worth it (?)

Reply:
:+1:

If you can use a constructor, great. If you can't, not big deal.